Updated the queries execution times, good performance overall

andremanzo1 opened the pull request.

huss commented:
@andremanzo1 Thank you for the updated performance info. I think it would be good if the original timing and the new timing were included rather than updating to the new timing. That way there is the record of the effect of the changes. Also, could the new code be run through explain and the website to pretty it so that is also added. Then it would be easier to see what happened with the changes. Thanks and let me know if you need anything.

andremanzo1 commented:
Hi Mr. Huss-Lederman

Would it be best to create a separate chart with the new timings?
Additionally, what would be the best way to organize this data?

When running the query through an EXPLAIN, would you prefer that I use the
same examples that Simon demonstrated previously?

Thank you.

On Mon, Mar 17, 2025 at 7:41 AM Steven Huss-Lederman <
***@***.***> wrote:

> @andremanzo1 <https://github.com/andremanzo1> Thank you for the updated
> performance info. I think it would be good if the original timing and the
> new timing were included rather than updating to the new timing. That way
> there is the record of the effect of the changes. Also, could the new code
> be run through explain and the website to pretty it so that is also added.
> Then it would be easier to see what happened with the changes. Thanks and
> let me know if you need anything.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/OpenEnergyDashboard/DesignDocs/pull/8#issuecomment-2729772214>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/A5URTKE52KJU5RNB65Y46NL2U3NKDAVCNFSM6AAAAABZEHWK56VHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMZDOMRZG43TEMRRGQ>
> .
> You are receiving this because you were mentioned.Message ID:
> ***@***.***>
> [image: huss]*huss* left a comment (OpenEnergyDashboard/DesignDocs#8)
> <https://github.com/OpenEnergyDashboard/DesignDocs/pull/8#issuecomment-2729772214>
>
> @andremanzo1 <https://github.com/andremanzo1> Thank you for the updated
> performance info. I think it would be good if the original timing and the
> new timing were included rather than updating to the new timing. That way
> there is the record of the effect of the changes. Also, could the new code
> be run through explain and the website to pretty it so that is also added.
> Then it would be easier to see what happened with the changes. Thanks and
> let me know if you need anything.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/OpenEnergyDashboard/DesignDocs/pull/8#issuecomment-2729772214>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/A5URTKE52KJU5RNB65Y46NL2U3NKDAVCNFSM6AAAAABZEHWK56VHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMZDOMRZG43TEMRRGQ>
> .
> You are receiving this because you were mentioned.Message ID:
> ***@***.***>
>

huss commented:
> Would it be best to create a separate chart with the new timings? 

That would be fine. It would also be fine to add column(s) to the current table with the new timings.

> Additionally, what would be the best way to organize this data? 

I'm not sure if I understand this one. The new data could be in new sections that mirror the current ones. Let me know if I didn't answer this one the way you want.

> When running the query through an EXPLAIN, would you prefer that I use the same examples that Simon demonstrated previously?

I think it would be best to redo one that had a full explain from before the changes. The results section has one for ``EXPLAIN (ANALYZE, BUFFERS) SELECT meter_3d_readings_unit ('{21}', 1, '2020-01-01 00:00:00', '2020-12-26 00:00:00', 2);`` that would be fine. The same formatting/info would be good. These are not the ones Simon did since that did not have it in pretty format.
